base/test/affinetransform_tests.cpp

```diff
@@ -0,0 +1,353 @@
+	int x;
+	int y;
+	float scale;
+	framemetadata_sp metadata;
+
+	AffineTestsStruct(const std::string& inpFilePath, int width, int height, ImageMetadata::ImageType imageType, int bit_depth, int angle, int x, int y, float scale)
+	{
+		if (imageType == ImageMetadata::ImageType::YUV420 || imageType == ImageMetadata::ImageType::YUV444) {
+			metadata = framemetadata_sp(new RawImagePlanarMetadata(width, height, imageType, size_t(0), CV_8U));
+		}
+		else {
+			metadata = framemetadata_sp(new RawImageMetadata(width, height, imageType, bit_depth, 0, CV_8U, FrameMetadata::HOST, true));
+		}
+		createPipeline(inpFilePath, width, height,  imageType, bit_depth,  angle,  x,  y,  scale);
+	}
+	void createPipeline(const std::string& inpFilePath, int width, int height, ImageMetadata::ImageType imageType, int bit_depth, int angle, int x, int y, float scale)
+	{
+		fileReader = boost::shared_ptr<FileReaderModule>(new FileReaderModule(FileReaderModuleProps(inpFilePath)));	
+		auto rawImagePin = fileReader->addOutputPin(metadata);
+
+		auto stream = cudastream_sp(new ApraCudaStream);
+		copy1 = boost::shared_ptr<Module>(new CudaMemCopy(CudaMemCopyProps(cudaMemcpyHostToDevice, stream)));
+		fileReader->setNext(copy1);
+
+		AffineTransformProps affineProps(stream, angle, x, y, scale);
+		affineTransform = boost::shared_ptr<Module>(new AffineTransform(affineProps));
+		copy1->setNext(affineTransform);
+		copy2 = boost::shared_ptr<Module>(new CudaMemCopy(CudaMemCopyProps(cudaMemcpyDeviceToHost, stream)));
+		affineTransform->setNext(copy2);
+		m3 = boost::shared_ptr<ExternalSinkModule>(new ExternalSinkModule());
+		copy2->setNext(m3);
+
+		BOOST_TEST(fileReader->init());
+		BOOST_TEST(copy1->init());
+		BOOST_TEST(affineTransform->init());
+		BOOST_TEST(copy2->init());
+		BOOST_TEST(m3->init());
+	}
+	boost::shared_ptr<FileReaderModule>fileReader;
+	boost::shared_ptr<Module>copy1;
+	boost::shared_ptr<Module>affineTransform;
+	boost::shared_ptr<Module>copy2;
+	boost::shared_ptr<ExternalSinkModule>m3;
+
+	~AffineTestsStruct()
+	{
+		m3->term();
+		copy2->term();
+		affineTransform->term();
+		copy1->term();
+		fileReader->term();
+	}
+};
+BOOST_AUTO_TEST_CASE(MONO_rotation, *boost::unit_test::disabled())
+{
+	ImageMetadata::ImageType::MONO;
+	AffineTestsStruct f("./data/mono_1920x1080.raw", 1920, 1080, ImageMetadata::ImageType::MONO, CV_8UC1, 5, 0, 0, 1.0f);
+
+	f.fileReader->step();
+	f.copy1->step();
+	f.affineTransform->step();
+	f.copy2->step();
+
+	auto outputPinId = f.copy2->getAllOutputPinsByType(FrameMetadata::RAW_IMAGE)[0];
+	auto frames = f.m3->pop();
+	BOOST_TEST((frames.find(outputPinId) != frames.end()));
+	auto outFrame = frames[outputPinId];
+	BOOST_TEST(outFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE);
+	Test_Utils::saveOrCompare("./data/testOutput/affinetransform_tests-MONO_rotation_mono_1920x1080.raw", (const uint8_t*)outFrame->data(), outFrame->size(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(MONO_scale, *boost::unit_test::disabled())
```

Comment:
why is this test disabled ?

Reply:
Many tests are disabled because each test will result in some result image. So, instead of having so many result images, we added tests where multiple transformations are done in one go. This way we were able to test the module but we do not have to have a lot of test result images in our repository.